src/components/auth/torus/AuthTorus.web.js

```diff
@@ -190,37 +237,34 @@ const AuthTorus = ({ screenProps, navigation, styles, store }) => {
         const { source } = await ready(replacing)
 
         log.debug('showing checkmark dialog')
+
         // showLoadingDialog(true)
         // await delay(30000000)
 
         // await new Promise(res => showLoadingDialog(true, res))
         // showLoadingDialog(true)
         // log.debug('hiding checkmark dialog')
         hideDialog()
-
-        //user exists reload with dashboard route
+        if (isSignUp) {
+          if (exists) {
+            return showEmailUsedDialog(provider, source)
```

Comment:
this logic is incorrect. you were supposed to check on server if email/mobile where used and with which provider, not the current user selected provider